internal/compile/conditions.go

```diff
@@ -124,6 +143,38 @@ func (ce *CELConditionEvaluator) Eval(input interface{}) (bool, error) {
 	return v, nil
 }
 
+// evaluateGlobals evaluates global values using std vars
+// then add calculated values to std vars and calculate expression.
+func (ce *CELConditionEvaluator) evaluateGlobals(stdvars map[string]interface{}) (map[string]ref.Val, error) {
+	values := make(map[string]ref.Val, len(ce.globals))
+	stdenv, err := cel.NewEnv(codegen.NewCELEnvOptions()...)
+	if err != nil {
+		return nil, err
+	}
+	for name, def := range ce.globals {
+		// TODO: Should we analyse condition expression to see if we even need to evaluate this def?
+		ast, issues := stdenv.Compile(def)
+		if issues != nil && issues.Err() != nil {
+			celLog.Warn("Global variable compilation failed", zap.Error(err))
+			return nil, issues.Err()
+		}
+		prg, err := stdenv.Program(ast)
+		if err != nil {
+			celLog.Warn("Global variable AST generation failed", zap.String(codegen.CELGlobalsIdent, name), zap.Error(err))
+			return nil, err
+		}
```

Comment:
What if we generate the program for globals at code generation time and store it in the `CELConditionEvaluator`? Then we don't have to do this compilation process each time the condition is evaluated.

Reply:
Yes, I think we can extend structures `codegen.Result` and `GeneratedPolicy` to include `globals map[string]*CELCondition` and `globals CelConditions map[string]*v1alpha1.CheckedExpr` respectively.